WordPress/Classes/ViewRelated/Media/CameraCaptureCoordinator.swift

```diff
@@ -16,14 +19,19 @@ final class CameraCaptureCoordinator {
         capturePresenter!.presentCapture()
     }
 
-    private func processMediaCaptured(_ mediaInfo: NSDictionary, blog: Blog) {
+    private func processMediaCaptured(_ mediaInfo: NSDictionary, blog: Blog, origin: UIViewController? = nil) {
         let completionBlock: WPMediaAddedBlock = { media, error in
             if error != nil || media == nil {
                 print("Adding media failed: ", error?.localizedDescription ?? "no media")
                 return
             }
-            guard let media = media as? PHAsset else {
-                    return
+            guard let media = media as? PHAsset,
+                    !media.exceedsFreeSitesAllowance(),
+                    !blog.hasPaidPlan else {
```

Comment:
I think this logic isn't correct. I'm seeing a case where media is exceeding the free sizes allowance, but the blog has a paid plan, but I'm not allowed to upload it. We only want to present the limit if both the media exceeds the limit and the blog doesn't have a paid plan.

Reply:
Thanks for catching this @frosty ! This is one more case where your suggestion to extract the `canUploadAsset` method comes in handy.